### ISSUE TITLE
Sample map browse not working and throws javascript errors

### ISSUE THREAD

**Issue by rija** [gigascience/gigadb-website#1944]
**Describe the bug**

**To Reproduce**
Steps to reproduce the behaviour:
1. go to https://gigadb.org (can be reproduced on staging at https://staging.gigadb.org if you don't have admin to live)
2. log in as an admin 
3. click on "Browse Samples"
4. no map is showing

**Expected behavior**
a map will show

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Safari and Firefox
 - Version: latest

**Additional context**

There is a javascript warning:
```
Loading failed for the <script> with source “https://cdn.jsdelivr.net/npm/ol@v8.1.0/dist/ol.js”. mapbrowse:187:65
```

followed by a javascript error:
```
Uncaught ReferenceError: ol is not defined
    <anonymous> https://staging.gigadb.org/site/mapbrowse:199
mapbrowse:199:16
```

There is also a javascript log entry:
```
Object { type: "FeatureCollection", features: [] }
​
features: Array []
​
type: "FeatureCollection"
​
<prototype>: Object { … }
mapbrowse:197:13
```

**Comment by only1chunts** [comment]
## Notes
These should effect the action of ticket but useful info to know:
- The map itself cannot be Google maps as that has restrictions within China, but I'm not precious as to which source map is used as long as it functions globaly.

- There is an extension to this bug in #831 where the map wont show much at the moment as its looking for the combined lat-lon attributes in the database. We stopped adding them as a combined value many years ago we now add them to the database as separate latitude and longitude attributes.

**Comment by rija** [comment]
Hi @only1chunts,

Regarding the second point, I believe this is out-dated information, as the problem you are describing has been fixed in this PR: https://github.com/gigascience/gigadb-website/pull/979

**Comment by rija** [comment]
We haven't changed the map provider, so we are not using Google map, so there shouldn't be any problem in that respect here.

**Comment by luistoptal** [comment]
@rija based on what I observe in staging (the map shows up okay in local), the issue might be related to Content Security Policy blocking loading map files from the cdn

Maybe need to add something along this lines here ops/configuration/nginx-conf/sites/nginx.target_deployment.https.conf.dist?

```
add_header Content-Security-Policy "default-src 'self'; style-src 'self' 'unsafe-inline' https://cdn.jsdelivr.net/npm/ol@v8.1.0/ol.css";
```

I'm not sure how to test this locally but I can spend some time looking into if it sounds like a good idea, please let me know

As for the console logs you mention, they are hardcoded in the view and should probably be removed

**Comment by kencho51** [comment]
Hi @rija,

When I executed the query in the `actionMapbrowse` in my staging database,
```
SELECT d.identifier,  d.title, satt.value, sp.scientific_name as sciname, s.id as sampleid FROM dataset as d
INNER JOIN dataset_sample as dsam on dsam.dataset_id = d.id
INNER JOIN sample as s on s.id = dsam.sample_id
INNER JOIN sample_attribute as satt on satt.sample_id=s.id
INNER JOIN species as sp on sp.id = s.species_id
where satt.attribute_id = 269 and d.upload_status='Published' order by sampleid;
```

the returned `value` are something like, `not recorded`, `00Ã‚Â°15' N 98Ã‚Â°08' E`, `off Heligoland, North Sea` or `31.48`. 

I then further looked into the sample_attribute table using:
```
SELECT attribute_id, value, count(*) FROM sample_attribute WHERE attribute_id = 269 GROUP BY attribute_id, value order by count desc;
```

And then got this returns:
```
attribute_id value count
269	not provided	3473
269	restricted access	1096
269	not recorded	257
269	not collected	131
269	32Â°35'N 103Â°37'E	74
269	30Â°0'N 103Â°02'E	38
269	31Â°04â€²37â€³N, 121Â°10â€²35â€³E	29
269	Multiple	27
269	Not Recorded	22
269	29Â°02'N 100Â°17'E	19
269	30Â°34'N 104Â°03'E	19
269	35Â°24'N 116Â°34'E	19
269	29Â°24'N 105Â°36'E	18
269	19Â°07â€²16â€³N, 109Â°05â€²00â€³E	11
269	42Â°20' N, 126Â°50' E	11
269	not applicable	10
269	31.48	9
269	N31Â°, E117Â°	9
269	35.9940Â° N, 78.8986Â° W	8
269	Aldabra, Indian Ocean	5
269	Sri Lanka, Indian Ocean	4
269	off Heligoland, North Sea	4
269	Morro Bay, San Luis Obispo, California, USA, Pacific Ocean	3
269	02Ã‚Â°50'56'' N 95Ã‚Â°56'24'' E	3
269	19Â°00â€²07â€³N, 109Â°32â€²33â€³E	3
269	Costa Brava, Spain, Mediterranean Sea	3
269	Kobe, Japan, Pacific Ocean	3
269	62Â°14'S, 58Â°47'W	3
269	29Â°24â€²00â€³N, 110Â°09â€²56â€³E	3
269	nor recorded	3
269	24Â°04â€²34â€³N, 110Â°08â€²17â€³E	3
269	Loblolly Bay, Anegada, British Virgin Islands, Caribbean Sea	2
269	00Ã‚Â°15' N 98Ã‚Â°07' E	2
269	00Ã‚Â°02' S 73Ã‚Â°24' E	2
269	23Â°20â€²22â€³N, 108Â°22â€²49â€³E	2
269	Weddell Sea, Southern Ocean	2
269	Caribbean Sea	2
269	Iceland Basin, Atlantic Ocean	2
269	37Â°26'N, 126Â°22'E	2
269	Atapupu, Timor, Indonesia, Indian Ocean	2
269	Singapore, Pacific Ocean	2
269	30.50′–30.180′N, 114.210′–114.390′E	2
269	Arendal, Norway, Skagerrak, North Sea	2
269	33Ã‚Â°35'20'' N 135Ã‚Â°10'50'' E	2
269	East Asia, Pacific Ocean	2
269	21Â°54â€²3â€³N, 111Â°23â€²58â€³E	2
269	13894; 19Â°33â€™3â€�N, 110Â°47â€™25â€� E	2
269	S off New Scotia, Atlantic Ocean	2
269	New Caledonia, Pacific Ocean	2
269	Bassin de la Gazelle, Kerguelen Islands, Southern Ocean	2
269	00Ã‚Â°15' N 98Ã‚Â°08' E	2
269	Gauss Base, Antarctica, Southern Ocean	2
269	18Â°59â€²58â€³N, 109Â°05â€²03â€³E	2
269	North Sea	2
269	off Florida, USA, Gulf of Mexico	2
269	not available	2
269	Penguin Bank, Oahu Island, Hawaii, USA, Pacific Ocean	2
269	35.202773	1
269	off Elephant Island, Southern Ocean	1
269	Monrovia, Liberia	1
269	24Â°58â€²12â€³N, 112Â°53â€²17â€³E	1
269	35.114876	1
269	23Â°36â€²21â€³N, 113Â°57â€²18â€³E	1
269	Abeloya, Kong Karls Land	1
269	Not recorded	1
269	30Â°36â€²28â€³N, 119Â°52â€²22â€³E	1
269	Romblon, Philippines, Pacific Ocean	1
269	Hastings, Barbados, Caribbean Sea	1
269	Dry Tortugas, Florida, USA, Gulf of Mexico	1
269	29Â°37â€²51â€³N, 109Â°50â€²17â€³E	1
269	Rovinj, Croatia, Mediterranean Sea	1
269	26Â°47â€²29â€³N, 117Â°02â€²21â€³E	1
269	Polana Beach, Mozambique, Indian Ocean	1
269	21Â°50â€²13â€³N, 108Â°03â€²27â€³E	1
269	Madeira, Portugal, Atlantic Ocean	1
269	Misaki, Japan, Pacific Ocean	1
269	Parry Island, Eniwetok Atoll, Marshall Islands, Pacific Ocean	1
269	18Â°54â€²40â€³N, 109Â°41â€²12â€³E	1
269	25Ã‚Â°40'00'' S 86Ã‚Â°46'05'' W	1
269	23Â°10â€²47â€³N, 113Â°21â€²33â€³E	1
269	18Â°39â€²43â€³N, 110Â°16â€²19â€³E	1
269	Pemba Channel, Tanzania, Indian Ocean	1
269	33Ã‚Â°56' N 26Ã‚Â°43' E	1
269	Port Jackson, Sydney, Australia, Pacific Ocean	1
269	Mombasa, Kenya, Indian Ocean	1
269	30Â°11â€²36â€³N, 118Â°09â€²20â€³E	1
269	35Ã‚Â°10' S 23Ã‚Â°02' E	1
269	Low Island, Low Isles, Australia, Pacific Ocean	1
269	01Ã‚Â°49' N 45Ã‚Â°29' E	1
269	Ecuador, Pacific Ocean	1
269	Ralum, East New Britain, Papua New Guinea, Pacific Ocean,	1
269	60Â° 9â€² 0â€³ N, 24Â° 39â€² 30â€³ E	1
269	Maroon Point, Mauritius, Indian Ocean	1
269	40Ã‚Â°0Ã‚Â´38.02Ã‚Â´'N3Ã‚Â°52Ã‚Â´41.4Ã‚Â´Ã‚Â´E	1
269	Paita, Peru, Pacific Ocean	1
269	St. Thomas, Caribbean Sea	1
269	22Â°26â€²46â€³N, 106Â°53â€²17â€³E	1
269	61Ã‚Â°42' N 09Ã‚Â°36' W	1
269	35.161870	1
269	29Â°46â€²34â€³N, 110Â°05â€²50â€³E	1
269	Golf of Suez, Red Sea, Indian Ocean	1
269	31Ã‚Â°58'82'' S 174Ã‚Â°15'87'' E	1
269	29Â°33â€²33â€³N, 103Â°18â€²45â€³E	1
269	Azores, Portugal, Atlantic Ocean	1
269	43Ã‚Â°56' S 60Ã‚Â°52' W	1
269	Papua New Guinea, Pacific Ocean	1
269	45.5Â° N, 122.7Â° W	1
269		1
269	06Ã‚Â°19' S 12Ã‚Â°02' E	1
269	18Â°45â€²15â€³N, 110Â°12â€²43â€³E	1
269	35.192253	1
269	Angra Pequena, LÃƒÂ¼deritz, Namibia, Atlantic Ocean	1
269	34.0689Â° N, 118.4452Â° W	1
269	Jakarta, Indonesia, Pacific Ocean	1
269	30Ã‚Â°02'09'' N 28Ã‚Â°24'02'' W	1
269	S Chile, Pacific Ocean	1
```

So clearly, besides the CSP problem, the `value` itself in the `sample_attribute` table is not standardized and not formatted well, it causes the `geojson_features` array is empty,  which leads to no `dots` can be seen in the map in the `/site/mapbrowse` page. 

The sanity check in the `mapbrowse.php` is not enough to check all the `value` cases in the `sample_attribute` table in the current production database.

**Comment by rija** [comment]
Closing as the fix to the CSP configuration is in PR #1977
